charts/fleet/templates/service_gitops_metrics.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if .Values.gitops.enabled }}
+{{- if .Values.metrics.enabled }}
+{{ $shards := list "" }}
+{{ if .Values.shards }}
+{{ $shards = concat $shards .Values.shards | uniq }}
+{{ end }}
+{{ range $shards }}
+apiVersion: v1
+kind: Service
+metadata:
+    name: "monitoring-gitops-controller{{if . }}-shard-{{ . }}{{end}}"
+    labels:
+        app: gitjob
+spec:
+    type: ClusterIP
+    ports:
+        - port: 8081
+          targetPort: 8081
+          protocol: TCP
+          name: metrics
+    selector:
+        app: gitjob
+        {{- if empty . }}
+        # fleet.cattle.io/shard-default: "true" 
```

Comment:
Doesn't that need to be enabled?

Reply:
It do not work for the moment as xavi didn't merge is PR

Reply:
Hmm, but it will work eventually and doesn't do any harm in the meantime. The same is true for `fleet.cattle.io/shard-id: <shard-id>`, right?

So, we either make it compatible with the sharding functionality that will be there in the future, or we just make it work for the unsharded gitjob controller. Both ways will enable us to proceed with this PR before https://github.com/rancher/fleet/pull/2537 is merged.

Personally, I'm fine with having that feature in already, even if it does not work yet, though, removing the sharding functionality from this controller and waiting for the sharding of gitjob controller is probably less controversial.

@weyfonk @0xavi0 

Any opinions? We need this service to be merged and those changes to be put into a release candidate to have QA testing the [PR in rancher/charts](https://github.com/rancher/charts/pull/4101), which depends on this service.